skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,185 @@
+        The length of the intensity vertex of the ellipsoid.
+    has_nan: bool, optional
+        If ``False`` (default) assumes that none of the values in ``image``
+        are ``np.nan``, and uses a faster implementation.
+
+    Returns
+    -------
+    filtered_image : ndarray
+        The image with background removed.
+
+    Notes
+    -----
+
+    - For the pixel that has its background intensity estimated (at ``(0,0)``)
+      the rolling ellipsoid method places an ellipsoid under it and
+      raises the ellipsoid until it touches the image umbra at ``pos=(y,x)``.
+      The background intensity is then estimated using the image intensity at
+      that position (``image[y, x]``) plus the difference of
+      ``intensity_vertex`` and the intensity of the ellipsoid at ``pos``. The
+      intensity of the ellipsoid is computed using the canonical ellipsis
+      equation::
+
+            semi_spatial = kernel_size / 2
+            semi_vertex = intensity_vertex / 2
+            np.sum((pos/semi_spatial)**2) + (intensity/semi_vertex)**2 = 1
+
+    - This algorithm assums that dark pixels correspond to the background. If
+      you have a bright background, invert the image before passing it to the
+      function, e.g., using `utils.invert`.
+    - This algorithm is sensitive to noise (in particular salt-and-pepper
+      noise). If this is a problem in your image, you can apply mild
+      gaussian smoothing before passing the image to this function.
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.morphology import rolling_ellipsoid
+    >>> result = rolling_ellipsoid(data.coins())
+    """
+
+    kernel_size = np.asarray(kernel_size)
+    if not np.issubdtype(kernel_size.dtype, np.number):
+        raise ValueError(
+            "kernel_size must be convertible to a numeric array.")
+    if not kernel_size.shape == (2,):
+        raise ValueError(
+            "kernel_size must be a two element tuple.")
+    if np.any(kernel_size <= 0):
+        raise ValueError("All elements of kernel_size must be greater zero.")
+    kernel_size = kernel_size / 2
+
+    intensity_vertex = np.asarray(intensity_vertex, dtype=np.float_)
+    if not intensity_vertex.shape == tuple():
+        raise ValueError("Intensity_vertex must be a scalar.")
+    if np.any(intensity_vertex <= 0):
+        raise ValueError("Intensity_vertex must be greater zero.")
+    intensity_vertex = intensity_vertex / 2
+
+    image = np.asarray(image)
+    img = image.astype(np.float_)
+
+    kernel_size_y, kernel_size_x = np.round(kernel_size).astype(int)
+
+    pad_amount = [(0, 0)] * image.ndim
+    pad_amount[-2] = (kernel_size_y, kernel_size_y)
+    pad_amount[-1] = (kernel_size_x, kernel_size_x)
```

Comment:
```suggestion
    pad_amount = [[0]] * (image.ndim - 2) + [[kernel_size_y], [kernel_size_x]]
```
`numpy.pad` will [internally expand `[[0], ..., [0], [y], [x]]` to `[[0, 0], ..., [0, 0 ], [y, y], [x, x]]`](https://github.com/numpy/numpy/blob/a2b9c2d5b6637b040917c0a2ef393dae83f09ee3/numpy/lib/arraypad.py#L454-L518).